src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonPropertiesView.ts

```diff
@@ -60,14 +71,31 @@ export class ExecutionPlanComparisonPropertiesView extends ExecutionPlanProperti
 			target = localize('executionPlanPropertiesEdgeOperationName', "Edge");
 		}
 
-		const titleText = localize('executionPlanComparisonPropertiesBottomOperation', "Bottom operation: {0}", target);
-		this._bottomOperationNameContainer.innerText = titleText;
-		this._bottomOperationNameContainer.title = titleText;
-		this.addDataToTable();
+		this._rightTitleText = localize('executionPlanComparisonPropertiesRightOperation', "Right operation: {0}", target);
+		this._bottomTitleText = localize('executionPlanComparisonPropertiesBottomOperation', "Bottom operation: {0}", target);
+		this._secondaryContainer.innerText = this._bottomTitleText;
+		this._secondaryContainer.title = this._bottomTitleText;
+		this.refreshPropertiesTable();
 	}
 
+	private updatePropertyContainerTitles(orientation: 'horizontal' | 'vertical'): void {
+		if (orientation === 'horizontal') {
+			this._primaryContainer.innerText = this._topTitleText;
+			this._primaryContainer.title = this._topTitleText;
+			this._secondaryContainer.innerText = this._bottomTitleText;
+			this._secondaryContainer.title = this._bottomTitleText;
+		}
+		else {
+			this._primaryContainer.innerText = this._leftTitleText;
+			this._primaryContainer.title = this._leftTitleText;
+			this._secondaryContainer.innerText = this._rightTitleText;
+			this._secondaryContainer.title = this._rightTitleText;
+		}
+
+		this.refreshPropertiesTable(orientation);
+	}
 
-	public addDataToTable() {
+	public refreshPropertiesTable(orientation: 'horizontal' | 'vertical' = 'vertical') {
```

Comment:
You should make a string union type for this orientation (or even better an enum) instead of defining it in multiple places like this.

Reply:
Is there a guideline that you follow for string types like this one? For example, once or twice is fine, but any more than that and switch to an enum or union type.

Reply:
The string union vs enum is mostly a personal preference, we do both. So use whichever one you prefer, I just find enums to be easier to use and something more people are familiar with.